src/Soneta.Sdk/Sdk/Sdk.targets

```diff
@@ -196,31 +196,41 @@
-      <WorkingXmls Include="@(BusinessXmls)" RemoveMetadata="$(MetadataToRemove)"/>
-      <WorkingXmls Include="@(ConfigXmls)" RemoveMetadata="$(MetadataToRemove)"/>
+      <WorkingXmls 
+        Include="@(None)" 
+        Condition="$([System.String]::new('%(Filename)%(Extension)').EndsWith('business.xml', StringComparison.InvariantCultureIgnoreCase ))"
+        RemoveMetadata="$(MetadataToRemove)" />
+      <WorkingXmls 
+        Include="@(EmbeddedResource)" 
+        Condition="$([System.String]::new('%(Filename)%(Extension)').EndsWith('config.xml', StringComparison.InvariantCultureIgnoreCase ))"
+        RemoveMetadata="$(MetadataToRemove)" />
     </ItemGroup>
   </Target>
 
-  <Target Name="ExecuteSonetaGenerator" BeforeTargets="CoreCompile" DependsOnTargets="PrepareSonetaGeneratorAssets"
-          Condition=" ( '$(RunSonetaGenerator)' == '' OR '$(RunSonetaGenerator)' != 'false' ) AND $( [System.IO.Directory]::GetFiles( '$(ProjectDir)' , '*.business.xml', SearchOption.AllDirectories ).Length ) &gt; 0 "
-          Inputs="%(WorkingXmls.Identity)" Outputs="@(WorkingXmls->'%(RelativeDir)%(Filename).cs')">
-
-    <ItemGroup>
-      <!-- Replace-hack służy do wyeliminowania pliku X.business.xml w chwili przetwarzania skojarzonego z nim X.config.xml -->
-      <SameAssemblyBusinessXmls Include="@(BusinessXmls)" Exclude="@(WorkingXmls->'%(Identity)'->Replace('.config.xml','.business.xml')->Replace('.Config.xml','.Business.xml'))"/>
-    </ItemGroup>
-
-    <Exec Condition="'$(SavedFile)' != '' AND '%(WorkingXmls.FullPath)' == '$(SavedFile)'"
-          Command="&quot;@(SonetaGeneratorExe)&quot; &quot;$(SavedFile)&quot; FILE" />
-    <Exec Condition="'@(SonetaSchemaReference->Count())' &gt; 0 AND '@(WorkingXmls->Count())' &gt; 0"
-          Command="&quot;@(SonetaGeneratorExe)&quot; -s &quot;@(SonetaSchemaReference,'&quot; &quot;')&quot; @(SameAssemblyBusinessXmls,' ') &quot;%(WorkingXmls.FullPath)&quot;" />
+  <Target Name="ExecuteSonetaGenerator" BeforeTargets="BeforeCompile;CoreCompile" DependsOnTargets="PrepareSonetaGeneratorAssets"
+          Condition="( '$(RunSonetaGenerator)' == '' OR '$(RunSonetaGenerator)' != 'false' ) 
+            AND (
+              $( [System.IO.Directory]::GetFiles( '$(ProjectDir)' , '*.business.xml', SearchOption.AllDirectories ).Length ) &gt; 0
+                OR 
+              $( [System.IO.Directory]::GetFiles( '$(ProjectDir)' , '*.config.xml', SearchOption.AllDirectories ).Length ) &gt; 0
+            )"
+          Inputs="@(WorkingXmls)" 
+          Outputs="%(RelativeDir)%(Filename).cs">
 
+    <PropertyGroup>
+      <GeneratedCSExists Condition="Exists('%(WorkingXmls.RelativeDir)%(WorkingXmls.Filename).cs')">true</GeneratedCSExists>
+    </PropertyGroup>
+    
+    <Exec Command="&quot;@(SonetaGeneratorExe)&quot; -s @(SonetaSchemaReference,'&quot; &quot;') &quot;%(WorkingXmls.FullPath)&quot;" />
```

Comment:
Z konstrukcji @(SonetaSchemaReference,'&quot; &quot;') zniknęły zewnętrzne &quot-y. W tym miejscu chcemy uzyskac liste sciezek ujetych w cudzyslow i rozdzielonych spacja. Brak zewnetrznych znakow psuje to rozwiazanie.

Reply:
Zdecydowałem się na inne, bardziej elastyczne rozwiązanie.